public_html/wp-content/plugins/wordcamp-speaker-feedback/includes/comment-meta.php

```diff
@@ -0,0 +1,207 @@
+		$schema['object_subtype'] = COMMENT_TYPE;
+
+		register_meta( 'comment', $key, $schema );
+	}
+}
+
+/**
+ * Define the properties of the feedback meta fields.
+ *
+ * @param string $key Optional. A specific key to get the schema for.
+ *
+ * @return array
+ */
+function get_feedback_meta_field_schema( $key = '' ) {
+	$schema = array(
+		'version' => array(
+			'description'       => 'The version of the feedback questions.',
+			'type'              => 'integer',
+			'single'            => true,
+			'sanitize_callback' => 'absint',
+			'show_in_rest'      => false,
+			'default'           => META_VERSION,
+			'required'          => true,
+		),
+		'rating'  => array(
+			'description'       => 'The rating. A number between 1 and 5.',
+			'type'              => 'integer',
+			'single'            => true,
+			'sanitize_callback' => 'absint',
+			'show_in_rest'      => false,
+			'default'           => 0,
+			'required'          => true,
+		),
+		'q1'      => array(
+			'description'       => 'The answer to the first feedback question.',
+			'type'              => 'string',
+			'single'            => true,
+			'sanitize_callback' => 'sanitize_text_field',
+			'show_in_rest'      => false,
+			'default'           => '',
+			'required'          => false,
+		),
+		'q2'      => array(
+			'description'       => 'The answer to the second feedback question.',
+			'type'              => 'string',
+			'single'            => true,
+			'sanitize_callback' => 'sanitize_text_field',
+			'show_in_rest'      => false,
+			'default'           => '',
+			'required'          => false,
+		),
+		'q3'      => array(
+			'description'       => 'The answer to the third feedback question.',
+			'type'              => 'string',
+			'single'            => true,
+			'sanitize_callback' => 'sanitize_text_field',
+			'show_in_rest'      => false,
+			'default'           => '',
+			'required'          => false,
+		),
+	);
+
+	if ( $key ) {
+		return $schema[ $key ] ?? array();
+	}
+
+	return $schema;
+}
+
+/**
+ * Check that an array of meta values has all required keys and contains valid data.
+ *
+ * @param array $meta
+ *
+ * @return array|WP_Error
+ */
+function validate_feedback_meta( $meta ) {
+	if ( ! isset( $meta['version'] ) ) {
+		$meta['version'] = META_VERSION;
+	}
+
+	$fields          = get_feedback_meta_field_schema();
+	$required_fields = array_filter(
+		wp_list_pluck( $fields, 'required' ),
+		function( $is_required ) {
+			return $is_required;
+		}
+	);
+
+	$missing_fields = array_diff_key( $required_fields, $meta );
+
+	if ( ! empty( $missing_fields ) ) {
+		return new WP_Error(
+			'feedback_meta_missing_field',
+			__( 'Please fill in all required fields.', 'wordcamporg' ),
+			array(
+				'missing_fields' => array_keys( $missing_fields ),
+			)
+		);
+	}
+
+	$integer_fields = array_keys( wp_list_pluck( $fields, 'type' ), 'integer', true );
+
+	foreach ( $integer_fields as $key ) {
+		if ( isset( $meta[ $key ] ) && ! is_numeric( $meta[ $key ] ) ) {
```

Comment:
Can we set some custom values in the schema? Thinking it might be good to check against a "max" so that no one tries to enter in a rating of 100 or something (in a future PR is fine)

Reply:
See what you think of this: 42180384